testcases/master/template-android.jinja2

```diff
@@ -0,0 +1,22 @@
+{% extends "testcases/master/template-master.jinja2" %}
+
+{% set OS_INFO = OS_INFO|default("android") %}
+{% set DEPLOY_OS = DEPLOY_OS|default("android") %}
+
+{# urls #}
+{% set DOCKER_IMAGE = DOCKER_IMAGE|default("yongqinliu/linaro-android-docker:0.1") %}
+{% set use_docker_image_for_test_target = use_docker_image_for_test_target|default(true) %}
+
+{# variables needs to be converted to be used by the root master.jinja2 #}
+{% set LAVA_JOB_PRIORITY = LAVA_JOB_PRIORITY|default("medium") %}
+
+{########## LAVA JOB DEFINITION ##############}
+{# override the job_name block defined in testcases/master/template-master.jinja2 #}
+{# with the BUILD_NUMBER in the middle, it's possible to get the sub job name #}
+{# from this lava job name information #}
+{% block job_name %}{{PROJECT_NAME}}-{{OS_INFO}}-{{KERNEL_BRANCH}}-{{BUILD_NUMBER}}-{{test_name}}{% endblock job_name %}
+
+{% block test_target %}
+    {{ super() }}
+    {% block test_target_definitions %}{% endblock test_target_definitions %}
```

Comment:
This isn't needed either I think.
I think we can redo this PR to something like http://ix.io/3ZaT
mkdir projects/lkft-android
projects/lkft-android/variables.yaml
"OS_INFO": "android"
"OS_DEPLOY": "android"
"DOCKER_IMAGE": "yongqinliu/linaro-android-docker:0.1"
"LAVA_JOB_PRIORITY": "medium"
"PROJECT": "projects/lkft-android/"
...

Reply:
Files deleted:
include/android-boottime-definition.jinja2
testcases/master/template-android.jinja2

Reply:
Can we please also add a test into
.github/workflows/test-plans-pipeline.yml

Reply:
> This isn't needed either I think. I think we can redo this PR to something like http://ix.io/3ZaT mkdir projects/lkft-android projects/lkft-android/variables.yaml "OS_INFO": "android" "OS_DEPLOY": "android" "DOCKER_IMAGE": "yongqinliu/linaro-android-docker:0.1" "LAVA_JOB_PRIORITY": "medium" "PROJECT": "projects/lkft-android/" ...

Just want to confirm, with your change, it needs to use_docker_image_for_test_target in every android jobs, like the cts, vts and others to be added later, and needs to specify the variables like OS_INFO, OS_DEPLOY etc everywhere, like in the variables files for this repository, and other places need to submit the tests. 
testcases/master/template-android.jinja2 is to avoid such problems and make it easier for users to define new android templates.
but if this is the style of lava-test-plans, then let's follow that rule.

Reply:
for testcases/master/template-test.jinja2, why the content is defined there but not in the testcases/master/template-master.jinja2 or include/test_target.jinja2, there seems too more fragments in the repository.

Reply:
> for testcases/master/template-test.jinja2, why the content is defined there but not in the testcases/master/template-master.jinja2 or include/test_target.jinja2, there seems too more fragments in the repository.

what I can see not all testcases files uses bits from template-test.jinja2 because its not applicable

Reply:
> > Files deleted: include/android-boottime-definition.jinja2 testcases/master/template-android.jinja2
> 
> This PR does not include any changes for the lkft-android project to be added later, will update .github/workflows/test-plans-pipeline.yml when the lkft-android project is added

but I'm confused, you add an testcase android-boot and that is a testcase that you can add to the pipeline.